File: img_pipe/plotting/plot_recon_anatomy.py
#plot_recon_anatomy.py
#
# Liberty Hamilton 2016
#

import mayavi
import scipy.io
import ctmr_brain_plot
from ../SupplementalFiles import FS_colorLUT
import numpy as np
import os

fs_dir = os.environ['SUBJECTS_DIR']

def plot_recon_anatomy(patient):
	subj = patient.subj
	hem = patient.hem
	a = scipy.io.loadmat('%s/%s/Meshes/%s_pial_trivert.mat'%(fs_dir, subj, hem))
	e = scipy.io.loadmat('%s/%s/elecs/TDT_elecs_all.mat'%(fs_dir, subj))

	# Plot the pial surface
	mesh, mlab = ctmr_brain_plot.ctmr_gauss_plot(a['tri'], a['vert'], color=(0.8, 0.8, 0.8))
	
	# Add the electrodes, colored by anatomical region
	elec_colors = np.zeros((e['elecmatrix'].shape[0], e['elecmatrix'].shape[1]))

	# Import freesurfer color lookup table as a dictionary
	cmap = FS_colorLUT.get_lut()

	# Make a list of electrode numbers
	elec_numbers = np.arange(e['elecmatrix'].shape[0])+1

	# Find all the unique brain areas in this subject
	brain_areas = np.unique(e['anatomy'][:,3])

	# Loop through unique brain areas and plot the electrodes in each brain area
	for b in brain_areas:
	    # Add relevant extra information to the label if needed for the color LUT
	    this_label = b[0]
	    if b[0][0:3]!='ctx' and b[0][0:4] != 'Left' and b[0][0:5] != 'Right' and b[0][0:5] != 'Brain' and b[0] != 'Unknown':
	        this_label = 'ctx-%s-%s'%(hem, b[0])
	        print(this_label)
	   	
	    if this_label != '':
	        el_color = np.array(cmap[this_label])/255.
	        ctmr_brain_plot.el_add(np.atleast_2d(e['elecmatrix'][e['anatomy'][:,3]==b,:]), 
	        					   color=tuple(el_color), numbers=elec_numbers[e['anatomy'][:,3]==b])
	mlab.show()        
	return mesh, mlab

File: img_pipe/plotting/matlab/plot_recon_anatomy.m
function [elecmatrix, anatomy] = plot_recon_anatomy(fs_dir, subjects_dir,subj, hem, elecfile_prefix, zero_indexed)
% [elecmatrix, anatomy] = plot_recon_anatomy(subj, hem)
% 
% Plots anatomy 

subj_dir = sprintf('%s/%s/', subjects_dir, subj);

load(sprintf('%s/Meshes/%s_%s_pial.mat', subj_dir, subj, hem));
load(sprintf('%s/elecs/%s.mat', subj_dir,elecfile_prefix));

figure;
c_h = ctmr_gauss_plot(cortex, [0 0 0], 0); 

if strcmp(hem, 'rh') %commented out b/c segfaulting...?
    loc_view(90,0);
end

brain_areas = unique(anatomy(:,4));
brain_areas(strcmp(brain_areas,'')) = []; %remove empty string segment from brain_areas
brain_areas(strcmp(brain_areas,'NaN')) = []; %remove empty string segment from brain_areas

% Make a jet color map with the same # of colors as brain areas
cm = colorcube(length(brain_areas));

% Loop through brain areas
for i=1:length(brain_areas)
    fprintf(1,'Brain area: %s\n', brain_areas{i});
    inds = find(strcmp(brain_areas{i}, anatomy(:,4)));
    if strcmp(zero_indexed,'True')
        inds_labels = inds-1;
    else
        inds_labels = inds;
    end
    % Have to change label for LUT
    if ~strcmp('CC_Anterior',strtrim(brain_areas{i})) && ~strcmp('CC_Mid_Anterior',strtrim(brain_areas{i})) && ~strcmp('Brain-Stem',strtrim(brain_areas{i})) && ~strcmp('WM',strtrim(brain_areas{i}(1:2))) && ~strcmp('ctx',brain_areas{i}(1:3)) && (length(brain_areas{i})<4 || ~strcmp('Left',brain_areas{i}(1:4))) && (length(brain_areas{i})<5 || ~strcmp('Right',brain_areas{i}(1:5)))
        this_area = sprintf('ctx-%s-%s', hem, brain_areas{i});
    else
        this_area = brain_areas{i};
    end

    grep_cmd = sprintf('grep ''%s'' %s/FreeSurferColorLUT.txt', this_area,fs_dir);
    [stat, result] = system(grep_cmd);

    if ~isempty(result)
        LUT = strsplit(result, '\s','DelimiterType','RegularExpression');
        elec_color = [str2double(LUT{3}) str2double(LUT{4}) str2double(LUT{5})]./255;
    else
        elec_color = cm(i,:);
    end
    
    el_add(elecmatrix(inds,:), 'elcol', elec_color, 'numbers', inds_labels, 'msize', 8);
    
end

leg = legend(['brain';brain_areas]);